src/components/formInput/FormInput.js

```diff
@@ -6,8 +8,22 @@ class FormInput extends Component {
     value: ''
   }
 
-  onShowForm = () => {
-    this.setState(prev => ({ print: !prev.print }));
+  static getDerivedStateFromProps(nextProps) {
+    if (nextProps.buttonClick) {
+      return { print: true };
+    }
+    if (!nextProps.buttonClick) {
+      return { print: false };
+    }
+    return null;
```

Comment:
why you don't just use directly `props.buttonClick`, why need id to keep in state?